neutone_sdk/cached_mel_spec.py

```diff
@@ -0,0 +1,130 @@
+import logging
+import os
+from typing import Optional, Callable
+
+import torch as tr
+from torch import Tensor
+from torch import nn
+from torchaudio.transforms import MelSpectrogram
+
+from neutone_sdk import CircularInplaceTensorQueue
+
+logging.basicConfig()
+log = logging.getLogger(__name__)
+log.setLevel(level=os.environ.get("LOGLEVEL", "INFO"))
+
+
+class CachedMelSpec(nn.Module):
+    def __init__(
+        self,
+        sr: int,
+        n_ch: int,
+        n_fft: int = 2048,
+        hop_len: int = 512,
+        f_min: float = 0.0,
+        f_max: Optional[float] = None,
+        n_mels: int = 128,
+        window_fn: Callable[..., Tensor] = tr.hann_window,
+        power: float = 2.0,
+        normalized: bool = False,
+        center: bool = True,
+        use_debug_mode: bool = True,
+    ) -> None:
+        """
+        Creates a Mel spectrogram that supports streaming of a centered, non-causal
+        Mel spectrogram operation that uses zero padding.
```

Comment:
It is obvious from the code, but could you specify here the amount of delay this adds?

Also the fact that it needs to receive hop_len*n sizes inputs.